src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -96,15 +111,17 @@ private FolderComposite FindInputFiles(string path, List<string> filesToExclude,
                 FullPath = Path.GetFullPath(path),
                 RelativePath = parentFolder is null ? lastPathComponent : Path.Combine(parentFolder, lastPathComponent)
             };
-            foreach (var folder in _fileSystem.Directory.EnumerateDirectories(folderComposite.FullPath).Where(x => !_foldersToExclude.Contains(Path.GetFileName(x))))
-            {
-                folderComposite.Add(FindInputFiles(folder, filesToExclude, folderComposite.RelativePath));
-            }
-            foreach (var file in _fileSystem.Directory.GetFiles(folderComposite.FullPath, "*.cs", SearchOption.TopDirectoryOnly))
+
+            // If directory actually exists, search it for files
+            if (_fileSystem.Directory.Exists(folderComposite.FullPath))
             {
-                // Roslyn cannot compile xaml.cs files generated by xamarin. 
-                // Since the files are generated they should not be mutated anyway, so skip these files.
-                if (!file.EndsWith(".xaml.cs"))
+                foreach (var folder in _fileSystem.Directory.EnumerateDirectories(folderComposite.FullPath).Where(x => !_foldersToExclude.Contains(Path.GetFileName(x))))
+                {
+                    folderComposite.Add(FindInputFiles(folder, filesToExclude, compileIncludeLinkedFiles, folderComposite.RelativePath));
+                }
+
+                // Find all cs files on disk except .xaml.cs files as they do not need to be compiled
+                foreach (var file in _fileSystem.Directory.GetFiles(folderComposite.FullPath, "*.cs", SearchOption.TopDirectoryOnly).Where(f => !f.EndsWith(".xaml.cs")))
```

Comment:
Same for getFiles, if we append the linked files to the files on the file system and return their real path they will be included correctly I guess.